api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,183 @@
+ * would span implementations should go on the {@link ConditionService}.
+ */
+public class ConditionServiceImplTest extends BaseContextSensitiveTest {
+	
+	protected static final String CONDITION_XML = "org/openmrs/api/include/ConditionServiceImplTest-SetupCondition.xml";
+	
+	private ConditionService conditionService;
+	
+	private PatientService patientService;
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+	
+	@Before
+	public void setup (){
+		if(conditionService == null){
+			conditionService = Context.getConditionService();
+		}
+		if(patientService == null){
+			patientService = Context.getPatientService();
+		}
+		executeDataSet(CONDITION_XML);
+	}
+
+	/**
+	 * @see ConditionService#saveCondition(Condition) 
+	 */
+	@Test
+	public void saveCondition_shouldSaveNewCondition(){
+		Integer patientId = 2;
+		Date date = new Date();
+		String uuid = "08002000-4469-12q3-551f-0339000c9a76";
+		String additionalDetail = "saving a condition";
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText();
+		Condition condition = new Condition();
+		condition.setCondition(codedOrFreeText);
+		condition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		condition.setAdditionalDetail(additionalDetail);
+		condition.setVerificationStatus(ConditionVerificationStatus.CONFIRMED);
+		condition.setOnsetDate(date);
+		condition.setUuid(uuid);
+		condition.setPatient(patientService.getPatient(patientId));
+		
+		conditionService.saveCondition(condition);
+		Condition savedCondition = conditionService.getConditionByUuid(uuid);
+		
+		Assert.assertEquals(additionalDetail, savedCondition.getAdditionalDetail());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, savedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, savedCondition.getClinicalStatus());
+		Assert.assertEquals(uuid, savedCondition.getUuid());
+		Assert.assertEquals(patientId, savedCondition.getPatient().getPatientId());
+		Assert.assertEquals(codedOrFreeText, savedCondition.getCondition());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldFindConditionGivenValidUuid() {
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		String clinical_status = "INACTIVE";
+		String verification_status = "PROVISIONAL";
+		Condition condition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(clinical_status, condition.getClinicalStatus().toString());
+		Assert.assertEquals(verification_status, condition.getVerificationStatus().toString());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getConditionById_shouldFindConditionGivenValidId() {
```

Comment:
Should this test be named `getConditionBy_shouldFindConditionGivenValidId(`)?

Reply:
It should be getCondition_shouldFindConditionGivenValidId() instead